sdk/containerregistry/Azure.Containers.ContainerRegistry/samples/Sample05_UploadDownloadImage.md

```diff
@@ -0,0 +1,160 @@
+// Download the manifest to obtain the list of files in the image
+DownloadManifestResult result = await client.DownloadManifestAsync(tag);
+OciImageManifest manifest = result.AsOciManifest();
+
+string manifestFile = Path.Combine(path, "manifest.json");
+using (FileStream stream = File.Create(manifestFile))
+{
+    await result.Content.ToStream().CopyToAsync(stream);
+}
+
+// Download and write out the config
+DownloadBlobResult configBlob = await client.DownloadBlobAsync(manifest.Config.Digest);
+
+string configFile = Path.Combine(path, "config.json");
+using (FileStream stream = File.Create(configFile))
+{
+    await configBlob.Content.ToStream().CopyToAsync(stream);
+}
+
+// Download and write out the layers
+foreach (OciBlobDescriptor layer in manifest.Layers)
+{
+    string layerFile = Path.Combine(path, TrimSha(layer.Digest));
+    using (FileStream stream = File.Create(layerFile))
+    {
+        await client.DownloadBlobToAsync(layer.Digest, stream);
+    }
+}
+
+static string TrimSha(string digest)
+{
+    int index = digest.IndexOf(':');
+    if (index > -1)
+    {
+        return digest.Substring(index + 1);
+    }
+
+    return digest;
+}
+```
+
+## Upload a custom manifest
+
+To upload an image with a custom manifest type, pass the `ManifestMediaType` to the `UploadManifest` method.
+
+```C# Snippet:ContainerRegistry_Samples_UploadCustomManifestAsync
+// Create a manifest file in the Docker v2 Manifest List format
+var manifestList = new
+{
+    schemaVersion = 2,
+    mediaType = ManifestMediaType.DockerManifestList.ToString(),
+    manifests = new[]
+    {
+        new
+        {
+            digest = "sha256:f54a58bc1aac5ea1a25d796ae155dc228b3f0e11d046ae276b39c4bf2f13d8c4",
+            mediaType = ManifestMediaType.DockerManifest.ToString(),
+            platform = new {
+                architecture = ArtifactArchitecture.Amd64.ToString(),
+                os = ArtifactOperatingSystem.Linux.ToString()
+            }
+        }
+    }
+};
+
+// Finally, upload the manifest file
+BinaryData content = BinaryData.FromObjectAsJson(manifestList);
+await client.UploadManifestAsync(content, tag: "sample", ManifestMediaType.DockerManifestList);
+```
+
+## Download a custom manifest
+
+To download a manifest of an unknown type, pass possible values to the list of acceptable media types.  The media type of the manifest is returned on the `DownloadManifestResult`.
+
+```C# Snippet:ContainerRegistry_Samples_DownloadCustomManifestAsync
+// Pass multiple media types if the media type of the manifest to download is unknown
+List<ManifestMediaType> mediaTypes = new() {
+    "application/vnd.docker.distribution.manifest.list.v2+json",
+    "application/vnd.oci.image.index.v1+json" };
+
+DownloadManifestResult result = await client.DownloadManifestAsync("sample", mediaTypes);
+
+if (result.MediaType == "application/vnd.docker.distribution.manifest.list.v2+json")
+{
+    Console.WriteLine("Manifest is a Docker manifest list.");
```

Comment:
nit: would be nice to show how to deserialize it from binary data

Reply:
Since we don't provide these types, I could show parsing the JSON - we also have an experimental type coming that will enable this, but it's not ready to include in production samples at present.